crates/ruff/src/checkers/tokens.rs

```diff
@@ -45,12 +45,15 @@ pub(crate) fn check_tokens(
         let mut state_machine = StateMachine::default();
         for &(ref tok, range) in tokens.iter().flatten() {
             let is_docstring = state_machine.consume(tok);
-            if matches!(tok, Tok::String { .. } | Tok::Comment(_)) {
+            if matches!(
+                tok,
+                Tok::String { .. } | Tok::FStringMiddle { .. } | Tok::Comment(_)
+            ) {
                 ruff::rules::ambiguous_unicode_character(
                     &mut diagnostics,
                     locator,
                     range,
-                    if tok.is_string() {
+                    if tok.is_string() || tok.is_f_string_middle() {
```

Comment:
Can fstrings be docstrings?

Reply:
No. Reference: https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals

> Formatted string literals cannot be used as docstrings, even if they do not include expressions.
> 
> ```python
> def foo():
>     f"Not a docstring"
> 
> foo.__doc__ is None
> # True
> ```

Reply:
I would then rewrite it for better clarity

```rust
if tok.is_string() && is_docstring {
	Context::Docstring
} else if tok.is_comment() {
	Context::Comment
} else {
	Context::String
}
```
or even using a match

```rust
let context = match tok {
	Tok::String => if is_docstring { Context::Docstring } else { Context::String },
	Tok::FStringMiddle => Context::String,
	Tok::Comment => Context::Comment,
	_ => unreachable!()
};

```